erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.py

```diff
@@ -0,0 +1,430 @@
+
+	total_debit, total_credit = 0,0
+	for d in data:
+		total_debit += flt(d.debit)
+		total_credit += flt(d.credit)
+
+	amounts_not_reflected_in_system = get_amounts_not_reflected_in_system(filters)
+
+	bank_bal = flt(balance_as_per_system) - flt(total_debit) + flt(total_credit) \
+		+ amounts_not_reflected_in_system
+
+	return bank_bal
+
+
+@frappe.whitelist()
+def update_bank_transaction(bank_transaction, reference_number, party_type=None, party=None):
+	# updates bank transaction based on the new parameters provided by the user from Vouchers
+	bank_transaction_name = bank_transaction
+	bank_transaction = frappe.get_doc("Bank Transaction", bank_transaction_name)
+	bank_transaction.reference_number = reference_number
+	bank_transaction.party_type = party_type
+	bank_transaction.party = party
+	bank_transaction.save()
+	return frappe.db.get_all('Bank Transaction',
+		filters={
+			'name': bank_transaction_name
+		},
+		fields=['date', 'deposit', 'withdrawal', 'currency',
+			'description', 'name', 'bank_account', 'company',
+			'unallocated_amount', 'reference_number',
+			 'party_type', 'party'],
+	)[0]
+
+
+@frappe.whitelist()
+def create_journal_entry_bts( bank_transaction, reference_number, reference_date, posting_date, entry_type,
+	# Create a new journal entry based on the bank transaction
+	second_account, mode_of_payment=None, party_type=None, party=None):
+	bank_transaction = frappe.get_doc("Bank Transaction", bank_transaction)
+	company_account = frappe.get_value("Bank Account", bank_transaction.bank_account, "account")
+	account_type = frappe.db.get_value("Account", second_account, "account_type")
+	if account_type in ["Receivable", "Payable"]:
+		if not (party_type and party):
+			frappe.throw(_("Party Type and Party is required for Receivable / Payable account {0}").format( second_account))
+	accounts = []
+	accounts.append({
+			"account": second_account,
+			"credit_in_account_currency": bank_transaction.deposit
+				if  bank_transaction.deposit > 0 
+				else 0,
+			"debit_in_account_currency":bank_transaction.withdrawal
+				if  bank_transaction.withdrawal > 0
+				else 0,
+			"party_type":party_type,
+			"party":party,
+		})
+
+	accounts.append({
+			"account": company_account,
+			"bank_account": bank_transaction.bank_account,
+			"credit_in_account_currency": bank_transaction.withdrawal
+				if  bank_transaction.withdrawal > 0
+				else 0,
+			"debit_in_account_currency":bank_transaction.deposit
+				if  bank_transaction.deposit > 0
+				else 0,
+		})
+
+	company = frappe.get_value("Account", company_account, "company")
+
+	journal_entry_dict = {
+		"doctype": 'Journal Entry',
+		"voucher_type" : entry_type,
+		"company" : company,
+		"posting_date" : posting_date,
+		"cheque_date" : reference_date,
+		"cheque_no" : reference_number,
+		"mode_of_payment" : mode_of_payment
+	}
+	journal_entry = frappe.get_doc(journal_entry_dict)
+	journal_entry.set("accounts", accounts)
+	journal_entry.insert()
+	journal_entry.submit()
```

Comment:
I just want to second casesolved-co-uk's request that if there's a way to create journal entries from bank transactions, which is a good thing, there needs to be a mode in which it only creates drafts, and does not automatically submit the journal entries.

Reply:
You can click on `edit in full page` to edit all the fields as per need

Reply:
I am confused; once a journal entry is submitted, it's permanently in the ledger and can only be reversed, not deleted. Some folks who don't even have the permissions to submit journal entries, only create them as drafts for a supervisor to review and submit, might well use this feature. Why the insistence on immediately submitting, when it might not actually be the desired Journal Entry after all? We strive to avoid erroneous journal entries to keep from cluttering our ledger with reversals, and careful and deliberate review is a key to that. Inserting and immediately submitting seems to be an obstacle to that, and against the spirit of why the draft and submitted stages were created. It seems to me that submitting should always be a manual operation, or at least ERPNext should support that mode for users who wish to operate that way. Thanks for your consideration!

Reply:
When edit in full page is clicked it does not insert or submit the entry
It opens the form view of the draft which can be edited and submitted.
There is a condition before these two lines which controls the behaviour